modules/opensearch/opensearch.client.inc

```diff
@@ -732,11 +425,130 @@ function opensearch_execute($request) {
+ *   The fast static cache used by opensearch_execute().
+ *
+ * @throws \TingClientException
+ *   If the search client request fails.
+ */
+function _opensearch_predict_cache(array $collections, array &$opensearch_static_fast) {
+  foreach ($collections as $collection) {
+    $object = reset($collection->objects);
+
+    $collection_request = opensearch_get_request_factory()->getCollectionRequest();
+    $collection_request->setAllObjects(FALSE);
+    $collection_request->setObjectId($object->id);
+    $collection_request->setQuery('*:*');
+
+    // See opensearch_execute() for description of always getting all relations.
+    $collection_request->setAllRelations(1);
+    $collection_request->setRelationData('full');
+
+    // Set agency from the administration interface.
+    if ($agency = variable_get('ting_agency', FALSE)) {
+      $collection_request->setAgency($agency);
+    }
+
+    // Set search profile from the administration interface.
+    $profile = variable_get('opensearch_search_profile', '');
+    if (!empty($profile)) {
+      $collection_request->setProfile($profile);
+    }
+
+    // Get additional parameters from other modules.
+    $params = module_invoke_all('opensearch_pre_execute', $collection_request);
+    if (!empty($params)) {
+      $collection_request->setParameters($params);
+    }
+
+    // Build the response that matches the request and put it into cache.
+    $collection_response = new TingClientObjectCollection($collection->objects);
+    _opensearch_cache_insert($collection_request, $collection_response, $opensearch_static_fast);
+
+    $object_request = opensearch_get_request_factory()->getObjectRequest();
+    $object_request->setObjectId($object->id);
+
+    // See opensearch_execute() for description of always getting all relations.
+    $object_request->setAllRelations(1);
+    $object_request->setRelationData('full');
+
+    // Set agency from the administration interface.
+    if ($agency = variable_get('ting_agency', FALSE)) {
+      $object_request->setAgency($agency);
+    }
+
+    // Set search profile from the administration interface.
+    $profile = variable_get('opensearch_search_profile', '');
+    if (!empty($profile)) {
+      $object_request->setProfile($profile);
+    }
+
+    // Build the response that matches the request and put it into cache.
+    $object_response = array();
+    $object_response[$object->id] = clone $object;
+    _opensearch_cache_insert($object_request, $object_response, $opensearch_static_fast);
+  }
+}
+
+/**
+ * Insert search response based on request.
+ *
+ * @param TingClientRequest $request
+ *   The search request that gives the response.
+ * @param \TingClientObject|\TingClientObjectCollection|array $response
+ *   The response to the request.
+ * @param array $opensearch_static_fast
+ *   The fast static cache used by opensearch_execute().
+ */
+function _opensearch_cache_insert(TingClientRequest $request, $response, array &$opensearch_static_fast) {
```

Comment:
If we have an `_opensearch_cache_insert` why is there no `_opensearch_cache_get`?

Reply:
Yes, the insert is used from with in different helper functions (trying to pre-heat/predict the cache) where get is only preformed in one place.